File: 221-maximal-square/maximal-square.cs
public class Solution {
    public int MaximalSquare(char[][] matrix) {
        if (matrix == null || matrix.Length == 0 || matrix[0].Length == 0) return 0;

        int m = matrix.Length;
        int n = matrix[0].Length;
        int maxSide = 0;
        
        int[,] dp = new int[m, n];
        
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                if (matrix[i][j] == '1') {
                    if (i == 0 || j == 0) {
                        dp[i, j] = 1;
                    } else {
                        dp[i, j] = Math.Min(Math.Min(dp[i-1, j], dp[i, j-1]), dp[i-1, j-1]) + 1;
                    }
                    maxSide = Math.Max(maxSide, dp[i, j]);
                }
            }
        }
        
        return maxSide * maxSide;
    }
}

File: 221-maximal-square/maximal-square.js
/**
 * @param {character[][]} matrix
 * @return {number}
 */
var maximalSquare = function(matrix) {
    if(!matrix.length) return 0;
    let m=matrix.length,n=matrix[0].length;
    let dp=Array(m).fill(null).map(()=>Array(n).fill(0));
    let maxSide=0;
    for(let i=0; i<m; i++){
        for (let j=0; j<n; j++){
            if(matrix[i][j]==='1'){
                if(i===0 || j===0){dp[i][j]=1;}
                else{dp[i][j]=Math.min(dp[i-1][j],dp[i][j-1],dp[i-1][j-1])+1;}
                maxSide=Math.max(maxSide,dp[i][j]);
            }
        }

    }
    return maxSide*maxSide;
};